### ISSUE TITLE
Please add statistics [Important]

### ISSUE THREAD

**Issue by davidhedlund** [uBlock-LLC/uBlock#1316]
In Adblock Plus is it possible to view "hit(s)" for each filter.

Do you plan to add this feature in uBlock?

**Comment by gorhill** [comment]
There is a resource cost to support this. Trying to be Adblock Plus is the wrong way to design uBlock. What matters is to solve real issues, not mimic Adblock Plus. The only rationale by OP is "ABP supports this", while instead a real problem should be spelled out, in which case, if the devs agree that there is an important issue to solve, they will come up with their own solution -- which may or may not be "hit counters".

**Comment by davidhedlund** [comment]
In uBlock I have no idea which of my custom filters that is being used or not which makes it hard to work with them. Manually compare them with the Request log is not an option, I have better things to do. The statistical feature in Adblock Plus is excellent. @chrisaljoudi what do you propose as solution to this problem?

**Comment by Gitoffthelawn** [comment]
Along similar lines, statistics allow the user to prune their rules (removing unused or outdated rules).  Also, by noticing specific rules are not be invoked, it alerts the user that the rule may need to be adjusted (requires a new matching pattern).

I don't find the way Adblock Plus handles it to be very good.  It appears very resource-intensive, and spreads the useful information across two columns: Last Hit, and # of Hits.  Each tells a piece of the story, but neither tells it fully.  A clever solution would ideally address the real tasks involved.

The way Adblock Plus handles it also creates serious privacy issues, although they are hopefully  confined to the user's local storage.  (Adblock Plus does include the option to disable this functionality, but by design cannot avoid the privacy issues when enabled.)

Bottom line: I agree with David that statistics are very useful and I agree with gorhill that what matters is to solve issues.  The big question is how to design and implement functionality to satisfy those issues.

I believe the first step is to identify the issues.  I've mentioned two above, and encourage others to add to the list.

**Comment by davidhedlund** [comment]
@Gitoffthelawn Thank you for pointing this out.

**Comment by jcosentino** [comment]
My concern is that, if a certain filter is blocking something that it's not supposed (false positive), I'm often re-adding ADP because I can easily view the specific filters in the respective subscriptions are causing the problem. I sort by "last hit" in order to quickly track them down.
Unless, there's a way to do this in uBlock that I have overlooked?

**Comment by davidhedlund** [comment]
@gorhill I receive this message when I'm trying to post an Issue in [your uBlock repository](https://github.com/gorhill/uBlock/): "You have been blocked from this repository."

I apologize if I have caused you any problems long ago, I do not remember why I'm blocked. Can you please consider to unblock me?

Regards.

**Comment by Gitoffthelawn** [comment]
@davidhedlund Gorhill blocked me as well.  I also have little idea why.  I'm guessing it was a control thing because he did it while I was helping out with the FAQ.

**Comment by davidhedlund** [comment]
@Gitoffthelawn Ok, I think gorhill get a lot of help if he unblocks all old blocked users. What do you think @gorhill?

**Comment by davidhedlund** [comment]
@Gitoffthelawn I moved this discussion to https://github.com/chrisaljoudi/uBlock/issues/1558

**Comment by davidhedlund** [comment]
This project is dead, use [uBlock Origin](https://github.com/gorhill/uBlock/issues) instead.

This issue is not relevant any more.
